insonmnia/hardware/hardware.go

```
@@ -146,56 +147,141 @@ func (m *benchValue) Value() uint64 {
 	return m.value
 }
 
-func insertBench(to []benchValue, bench *sonm.Benchmark, proportion float64) ([]benchValue, error) {
+func insertBench(to map[uint64]*sonm.Benchmark, bench *sonm.Benchmark, proportion float64) {
 	if math.IsNaN(proportion) || math.IsInf(proportion, 0) {
 		proportion = 0.0
 	}
+
 	id := bench.GetID()
-	for len(to) <= int(id) {
-		to = append(to, benchValue{})
+	_, wasSet := to[id]
+	if !wasSet {
+		to[id] = deepcopy.Copy(bench).(*sonm.Benchmark)
 	}
-
-	result := bench.GetResult()
+	target := to[id]
 	switch bench.SplittingAlgorithm {
 	case sonm.SplittingAlgorithm_NONE:
-		if to[id].IsSet() {
-			return nil, fmt.Errorf("duplicate benchmark with id %d and type none", bench.ID)
+		if wasSet {
+			panic(fmt.Errorf("duplicate benchmark with id %d and type none", bench.ID))
```

why exactly do we panic here (instead of returning an error)?